### PR TITLE
BUG: Changing way of applying numpy functions on Series

### LINKED ISSUE #33492: BUG: apply for series of numpy arrays falis with np functions
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python
import pandas as pd
import numpy as np

df = pd.DataFrame({"col": [np.array([.1, .2], dtype=float)]})
np.sin(df['col'].iloc[0])   # This works fine
df["col"].apply(np.sin)     # This breaks
```
Raises:

```python
AttributeError: 'numpy.ndarray' object has no attribute 'sin'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "~/main.py", line 6, in <module>
    df["col"].apply(np.sin)
  File "~/anaconda3/envs/pandas-issue/lib/python3.7/site-packages/pandas/core/series.py", line 3840, in apply
    return f(self)
  File "~/anaconda3/envs/pandas-issue/lib/python3.7/site-packages/pandas/core/series.py", line 679, in __array_ufunc__
    result = getattr(ufunc, method)(*inputs, **kwargs)
TypeError: loop of ufunc does not support argument 0 of type numpy.ndarray which has no callable sin method
```

#### Problem description

When I have a pd.Series where each row is a numerical numpy array, I can't apply numpy functions to them, I tried np.sin, np.cos, np.exp, but I think it's a general problem. I can apply them to individual rows manually and it works fine.

Moreover, If I import tqdm with tqdm.pandas and use progress_apply instead of apply it also works just fine!

#### Expected Output
```
0    [0.09983341664682815, 0.19866933079506122]
Name: col, dtype: object
```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.6.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.3.0-46-generic
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.0.3
numpy            : 1.18.1
pytz             : 2019.3
dateutil         : 2.8.1
pip              : 20.0.2
setuptools       : 46.1.3.post20200325
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
pytest           : None
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by mproszewska**
- [x] closes #33492 
- [x] tests added and passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] numpy functions on Series are called like other functions

### CODE REVIEW

**Comment by mproszewska** [comment]
Building documentation returns
```
/home/runner/miniconda3/envs/pandas-dev/lib/python3.8/site-packages/IPython/sphinxext/ipython_directive.py:1023: UserWarning: Code input with no code at /home/runner/work/pandas/pandas/doc/source/user_guide/computation.rst, line 622
```
which looks like issue [#26788](https://github.com/pandas-dev/pandas/issues/26788)

**Comment by TomAugspurger** [comment]
Is this going to kill performance for something like the following?

```python
s = pd.Series(list(range(100_000))
s.apply(np.sin)
```

File: pandas/tests/series/test_apply.py [hunk 1]
```diff
@@ -812,3 +812,11 @@ def test_apply_to_timedelta(self):
         b = Series(list_of_strings).apply(pd.to_timedelta)  # noqa
         # Can't compare until apply on a Series gives the correct dtype
         # assert_series_equal(a, b)
+
+    def test__apply_to_numpy_func(self):
```
Comment (jbrockmendel) [inline pandas/tests/series/test_apply.py:816]: no double-underscore

**Comment by mproszewska** [comment]
> Is this going to kill performance for something like the following?
> 
> ```python
> s = pd.Series(list(range(100_000))
> s.apply(np.sin)
> ```
I think so, but I feel like it's more appropriate use of the `apply` function. I think that using numpy functions in the old way should be possible but not like that.

File: pandas/core/series.py [hunk 2]
```diff
@@ -3847,9 +3847,6 @@ def f(x):
             f = func
 
         with np.errstate(all="ignore"):
-            if isinstance(f, np.ufunc):
```
Comment (jreback) [inline pandas/core/series.py:3850]: why would we want to do this?
is going to kill perf

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/series.py (hunk 2 above)
Comment (mproszewska) [reply pandas/core/series.py:3850]: Issue #33492 describes that. I also added test

Comment (jreback) [reply pandas/core/series.py:3850]: as I said I am not inclided to accept this w/o proof that this does not kill perf. We likely don't have asv's for this (or even using a np.ufunc in apply).

Now if you add those and either perf is not degraded, or you actually catch the TypeError then I would be ok.

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by mproszewska** [comment]
I felt like numpy functions could be called differently e.g. np.sin(series). 
I could add `if` that checks if series dtype is numpy array.
However, I wouldn't expect that numpy array would be used as dtype, I would usually expect inferred object dtype.

**Comment by TomAugspurger** [comment]
If we're going to change this then we'll need to deprecate the current behavior first. I haven't looked closely at the original issue, but at a glance it looks like it's dealing with nested data, which isn't especially well-supported in pandas. It's not clear to me that the gain from changing the implementation here is worth the effect to existing users.

**Comment by mproszewska** [comment]
Maybe better solution would be adding new parameter e.g if parameter force_mapping is set to True, function will be applied on each element separately.

**Comment by TomAugspurger** [comment]
I don't like just adding the `force_mapping` keyword. It's not clear from the name why you would want it.

*If* we want to do this, I think the solution is to actually deprecate the current behavior of ufuncs in `apply` and instruct people to use `.pipe(ufunc)` rather than `.apply(ufunc)`. But I'm not sure right now if it's worth the churn.

**Comment by jreback** [comment]
I agree with @TomAugspurger here, would be ok with showing a `PerformanceWarning`, but unlikely to be able to change / deprecate this behavior and cannot kill performance like this.

**Comment by mproszewska** [comment]
What do you mean by showing ``PerformanceWarning``? Where should be this warning included?
Should I provide better parameter name and description? or are you completely against idea behind that PR?

**Comment by jreback** [comment]
> What do you mean by showing `PerformanceWarning`? Where should be this warning included?
> Should I provide better parameter name and description? or are you completely against idea behind that PR?

we are -1 on adding any options to this function. you cannot simply *change* the way something like this works w/o showing the perf benefit (or drag in this case). I suspect this is likely going to be bad.

So the only way this could possibly go forward is to detect cases where this is likey to be used (e.g. using a ufunc on an object dtype), then dispatch that way. You might need to then show a `PerformanceWarning` in that case.

**Comment by TomAugspurger** [comment]
@jreback do you have a preference for `PerformanceWarning` vs. deprecating the current behavior. IMO, deprecating the current behavior is the best option: We *don't* slow down people's code immediately. Gives them time to update their code from `.apply(np.sin)` to `.pipe(np.sin)`.

The downside is that we can't fix the bug immediately. But IMO not slowing down this code without warning is more important than fixing the bug that (I think) only shows up for nested data.

**Comment by jreback** [comment]
I don’t think we should change this at all including deprecation

**Comment by jreback** [comment]
closing this as I don't see positions changing here. If you want to raise an issue and get consensus on a change okie dokie.

**Pull request closed without merging**